pkg/queue/metrics.go

```diff
@@ -24,6 +24,23 @@ type SchedulerMetricsType struct {
 	ErrorCounter    *prometheus.CounterVec
 }
 
+// WorkerMetricsType provides access to the prometheus metric objects for a worker
+type WorkerMetricsType struct {
+	Labels []string
+
+	ActiveGauge     prometheus.Gauge
+	WorkingGauge    prometheus.Gauge
+	WaitingGauge    prometheus.Gauge
+	DequeueingGauge prometheus.Gauge
+
+	ProcessingDuration prometheus.Histogram
+	DequeueingDuration prometheus.Histogram
+
+	ProcessedCounter    *prometheus.CounterVec
+	DequeueErrorCounter prometheus.Counter
+	ErrorsCounter       prometheus.Counter
```

Comment:
Why did you change most of these from the vector metrics to the plain kind? For example `ErrorsCounter` from `CounterVec` to `Counter`? This means we can add labels for queue or task type on the errors. It seems that almost all of the metrics will no longer support. Most of the work metrics were previously `*Vec` metrics so that we could track which queue and task type. Moving tot he plain counter radically reduces what is monitorable. I was just telling chris that we can actually monitor and alert of sync errors, or the future datasource heartbeat error, but that will no longer be possible.

Reply:
I changed it in places where the queue name and task type are not available for labeling at all.